services/api/service.go

```diff
@@ -950,14 +1123,20 @@ func (api *RelayAPI) handleSubmitNewBlock(w http.ResponseWriter, req *http.Reque
 		}
 	}
 
-	builderIsHighPrio, builderIsBlacklisted, err := api.redis.GetBlockBuilderStatus(payload.Message.BuilderPubkey.String())
+	builderPubkey := payload.Message.BuilderPubkey.String()
+	builderEntry, ok := api.blockBuildersCache[builderPubkey]
+	if !ok {
+		log.Warnf("unable to read builder: %x from the builder cache, using low-prio and no collateral", builderPubkey)
+		builderEntry = &blockBuilderCacheEntry{
+			status: common.BuilderStatus{
+				IsHighPrio: false,
+			},
+			collateral: ZeroU256,
+		}
+	}
```

Comment:
I'm tempted to error out and `return`. The reason is that this path should never happen. And if it does, alerting the builder instead of silently failing may be better. It's also fewer lines of code :)

Reply:
so the builder won't be in the DB until the first call to `UpsertBlockBuilderEntryAfterSubmission` which takes place after this (line 1248 in current rev). since the cache reads from the DB, if we don't have this, then the builder will never have a way to get added to the table. 

the builder status and collateral also can't be modified until the builder is in the table. we could add an endpoint to manually add the builder, but this would add complexity. IMO logging with warning here feels correct, but happy to continue discussing if you still think we should change it!

Reply:
Is there an equivalent of `log.Warnf` but as an error?

Reply:
yes! there is `log.Errorf` i didn't use this because every new builder that connects will hit this code path, so it could potentially be a lot of noise in terms of errors that are actually expected. happy to do errorf if you think it is better though!